packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3287,9 +3287,31 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @mustCallSuper
   void didChangeDependencies() {
     assert(_active); // otherwise markNeedsBuild is a no-op
+    _debugCheckOwnerBuildTargetExists('didChangeDependencies');
```

Comment:
these calls should be wrapped in asserts so that they don't cost anything in release builds